src/go/k8s/pkg/resources/resource.go

```diff
@@ -1,16 +1,23 @@
-package resources
+// Copyright 2021 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
 
-import (
-	"context"
+// Package resources contains reconciliation logic for redpanda.vectorized.io CRD
+package resources
 
-	"k8s.io/apimachinery/pkg/types"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-)
+import "context"
 
+// Resource decompose the reconciliation loop to specific kubernetes obejcts
 type Resource interface {
-	Obj() (client.Object, error)
```

Comment:
Why are these removed from the interface? Does this mean we can have a valid resource without a key or obj function?

Reply:
Yes, `Key` and `Obj` is used only inside the `Resource.Ensure` function. I should comment that earlier when the interface was introduced. I will bring back this function and `Key` too with the proper comment. My proposal:
> Obj returns resource managed client.Object

After some thought `Obj` is too generic. Client of such interface will need to know what is managed in the underlying resource.